app/models/concerns/reviewable_action_builder.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module ReviewableActionBuilder
+  extend ActiveSupport::Concern
+
+  # Build a single reviewable action and add it to the provided actions list.
+  # This is the canonical API used by both the legacy and refreshed UI code paths.
+  #
+  # @param actions [Reviewable::Actions] Actions instance to add to.
+  # @param id [Symbol] Symbol for the action, used to derive I18n keys.
+  # @param icon [String] Optional name of the icon to display with the action. Ignored in the refreshed UI.
+  # @param button_class [String] Optional CSS class for buttons in clients that render it.
+  # @param bundle [Reviewable::Actions::Bundle] Optional bundle object returned by add_bundle to group actions.
+  # @param client_action [String] Optional client-side action identifier (e.g. "edit").
+  # @param confirm [Boolean] When true, uses "reviewables.actions.<id>.confirm" for confirm_message.
+  # @param require_reject_reason [Boolean] When true, requires a rejection reason for the action.
+  #
+  # @return [Reviewable::Actions] The updated actions instance.
+  def build_action(
+    actions,
+    id,
+    icon: nil,
+    button_class: nil,
+    bundle: nil,
+    client_action: nil,
+    confirm: false,
+    require_reject_reason: false
+  )
+    actions.add(id, bundle: bundle) do |action|
+      prefix = "reviewables.actions.#{id}"
+      action.icon = icon if icon
+      action.button_class = button_class if button_class
+      action.label = "#{prefix}.title"
+      action.description = "#{prefix}.description"
+      action.client_action = client_action if client_action
+      action.confirm_message = "#{prefix}.confirm" if confirm
```

Comment:
This change assumes that all actions have a `reviewables.actions.#{id}.confirm` i18n key but that doesn't seem to be the case for all actions.

Reply:
Looking further down, I realised this pattern is not new and us used by `ReviewableFlaggedPost` and `ReviewablePost`.  Will resolve this for now.

Reply:
Yah, it's pre-existing logic. I've documented it in the method rdoc, I don't think an extra safety check is necessary here.